File: leetsync/3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.java
class Solution {
    public int countValidSelections(int[] nums) {
        int n = nums.length;
        int totalSum = 0, result = 0;
        for (int num : nums) {
            totalSum += num;
        }
        int currSum = 0;
        for (int i = 0; i < n; i++) {
            currSum += nums[i];
            int leftSum = currSum;
            int rightSum = totalSum - leftSum;
            if (nums[i] != 0) {
                continue;
            }
            if (leftSum == rightSum) {
                result += 2;
            } else if (Math.abs(leftSum - rightSum) == 1) {
                result++;
            }
        }
        return result;
    }
}

File: leetsync/3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.rs
impl Solution {
    pub fn count_valid_selections(nums: Vec<i32>) -> i32 {
        
        let mut total_sum = 0;
        let mut curr_sum = 0;
        let mut result = 0;
        let n = nums.len();
        for num in nums.iter() {
            total_sum += num;
        }
        for i in 0..n {
            curr_sum += nums[i];
            let mut left_sum = curr_sum;
            let mut right_sum = total_sum - curr_sum;
            if nums[i] != 0 {
                continue;
            }
            if left_sum == right_sum {
                result += 2;
            } else if (left_sum - right_sum).abs() == 1 {
                result += 1;
            }
        }
        return result;
    }
}